flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNSessionFIFOITCase.java

```diff
@@ -152,33 +154,32 @@ void runDetachedModeTest(Map<String, String> securityProperties) throws Exceptio
 
         jobRunner.join();
 
+        final long timeoutInSecs = 20;
+        final long testConditionIntervalInMillis = 500;
         // in "new" mode we can only wait after the job is submitted, because TMs
         // are spun up lazily
-        LOG.info("Waiting until two containers are running");
         // wait until two containers are running
-        while (getRunningContainers() < 2) {
-            sleep(500);
-        }
+        LOG.info("Waiting until two containers are running");
+        CommonTestUtils.waitUntilCondition(
+                () -> getRunningContainers() == 2,
```

Comment:
maybe >= 2 to be on the safe side?

Reply:
I thought the same but decided differently because I having more than 2 containers would be, afaiu, be an invalid state as well. Would you agree?

Reply:
well yes, but now this test is inconsistent. It can start 100 containers if it wants; if the condition runs at the right time it will pass nonetheless.

Reply:
Fair enough. I addressed your comment updating the condition to `>= 2`.